src/utils/encoding.ts

```diff
@@ -15,11 +15,11 @@ export const generateId = (source: string) => {
     idString += `?${firstParam[0]}=${firstParam[1]}`;
   }
 
-  return encodeURIComponent(Buffer.from(idString).toString('base64'));
+  return Buffer.from(idString).toString('base64url');
 };
 
 export const getSourceFromId = (id: string) => {
-  const decoded = Buffer.from(decodeURIComponent(id), 'base64').toString('utf8');
+  const decoded = Buffer.from(id, 'base64url').toString('utf8');
 
   return `https://${decoded}`;
 };
```

Comment:
To prevent broken links, this must remain backward-compatible:
```ts
export const getSourceFromId = (id: string) => {
  let decoded: string;
  try {
    decoded = Buffer.from(id, 'base64url').toString('utf8');
  } catch {
    // TEMPFIX: Handle old URI-encoded Base64
    const raw = decodeURIComponent(id);
    decoded = Buffer.from(raw, 'base64').toString('utf8');
  }
  return `https://${decoded}`;
};
```

Reply:
Ah true.

Reply:
makes sense, let's give this a try